airflow/utils/task_group.py

```diff
@@ -453,6 +454,22 @@ def topological_sort(self, _include_subdag_tasks: bool = False):
 
         return graph_sorted
 
+    def get_task_dict(self) -> dict[str, AbstractOperator]:
+        """Returns a flat dictionary of task_id: AbstractOperator"""
+        task_map = {}
+        groups_to_visit = [self]
+
+        while groups_to_visit:
+            visiting = groups_to_visit.pop(0)
+
+            for child in visiting.children.values():
+                if isinstance(child, AbstractOperator):
+                    task_map[child.task_id] = child
+                elif isinstance(child, TaskGroup):
```

Comment:
`elif` just in case `DAGNode` get additionnal subclasses

Reply:
I’d add an additional `else` block below and throw a ValueError or something to make sure a new class does not get ignored accidentally. Also if you annotate `task_map` this would be handled by Mypy.

Reply:
This is much better, thanks.

Done :)